src/components/encoding-pane/field-customizer.tsx

```diff
@@ -12,35 +14,136 @@ export interface FieldCustomizerProps extends ActionHandler<SpecEncodingAction>
   fieldDef: ShelfFieldDef;
 }
 
+export interface CustomProp {
+  prop: string;
+  nestedProp: string;
+}
 
-export class FieldCustomizerBase extends React.PureComponent<FieldCustomizerProps, {}> {
+const POSITIONAL_ENCODING_QUANTITATIVE = {
+  'Common': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'stack',
+      nestedProp: 'stack'
+    }
+  ],
+  'Scale': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Axis': [
+    {
+      prop: 'axis',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    }
+  ],
+  'Stack': [
+    {
+      prop: 'stack',
+      nestedProp: 'stack'
+    }
+  ]
+};
 
+const POSITIONAL_ENCODING_NON_QUANTITATIVE = {
+  'Scale': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Axis': [
+    {
+      prop: 'axis',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    }
+  ]
+};
+
+// TODO: Mark channel encoding
+const CHANNEL_ENCODING_MAP = {
```

Comment:
This name is also NOT descriptive.  

At suggested earlier, having a map/index here will restrict the type of properties that affect the field property index.  Why don't you just have switch-case or if-else inside the `getEncodingTypeForChannel` method (the method should be renamed though)?